src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1038,6 +1038,96 @@ public virtual ForbidResult Forbid(IList<string> authenticationSchemes, Authenti
             => new ForbidResult(authenticationSchemes, properties);
 
         /// <summary>
+        /// Creates a <see cref="SignInResult"/> with the specified authentication scheme.
+        /// </summary>
+        /// <param name="authenticationScheme">The authentication scheme to use for the sign-in operation.</param>
+        /// <param name="principal">The <see cref="ClaimsPrincipal"/> containing the user claims.</param>
+        /// <returns>The created <see cref="SignInResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignInResult SignIn(string authenticationScheme, ClaimsPrincipal principal)
+            => new SignInResult(authenticationScheme, principal);
+
+        /// <summary>
+        /// Creates a <see cref="SignInResult"/> with the specified authentication schemes.
+        /// </summary>
+        /// <param name="authenticationSchemes">The authentication schemes to use for the sign-in operation.</param>
+        /// <param name="principal">The <see cref="ClaimsPrincipal"/> containing the user claims.</param>
+        /// <returns>The created <see cref="SignInResult"/> for the response.</returns>
+        [NonAction]
+        public virtual SignInResult SignIn(IList<string> authenticationSchemes, ClaimsPrincipal principal)
```

Comment:
Okay for `SignIn` but for `SignOut`, see https://github.com/aspnet/Mvc/pull/4259#discussion_r55586374.